content/chapters/software-stack/lab/content/basic-syscall.md

```diff
@@ -3,6 +3,16 @@
 To get a better grasp on how the software stack works, let's do a bottom-up approach:
 we build and run different programs, that start of by using the system call API (the lowest layer in the software stack) and progressively use higher layers.
 
+### System Calls Explained
+
+A system call, or syscall for short, is a method used by applications to communicate with the operating system's kernel.
+
+The need for syscalls is tied to the modern operating systems model of conceptually separate into kernel space and user space. The kernel space is responsible for managing the hardware resources and provide an interface for the user space applications to request information from the hardware. The user space is where you are running your applications and processes.
```

Comment:
We should specify what are those hardware resources: CPU, memory, and I/O devices. Also, better than saying "request information" we could add some examples to highlight what a syscall enable the user to do:
- brk() is used to allocate memory
- getpid() ...
- open() ...

Reply:
+1